bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -351,28 +350,33 @@ func (ms *ManagedStream) Close() error {
 
 	var arc *storagepb.BigQueryWrite_AppendRowsClient
 
-	// Critical section: get connection, close, mark closed.
-	ms.mu.Lock()
-	arc, ch, err := ms.getStream(arc, false)
-	if err != nil {
-		return err
-	}
-	if ms.arc == nil {
-		return fmt.Errorf("no stream exists")
-	}
-	err = (*arc).CloseSend()
-	if err == nil {
+	closeErr := func() error {
```

Comment:
Since this part is not run in a loop, I don't see the need for a closure; could you just `defer ms.mu.Unlock()`? That would also protect the following block from races

```
	if ms.cancel != nil {
		ms.cancel()
		ms.cancel = nil
	}
```

Reply:
Simplified the Close().